react/components/SponsoredBanner/SponsoredBanner.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import { Spinner } from 'vtex.styleguide'
+import { useCssHandles } from 'vtex.css-handles'
+
+import { useSponsoredBanner } from './hook'
+import { getDataProperties } from '../../utils'
+import type { SponsoredBannersProps } from '../../interfaces'
+
+import './styles.css'
+
+const CSS_HANDLES = ['bannerWrapper', 'bannerImage'] as const
+
+export const SponsoredBanner = (props: SponsoredBannersProps) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const {
+    sponsoredBanners,
+    error,
+    handleClick,
+    loading,
+    styleProps,
+    imageAlt,
+  } = useSponsoredBanner(props)
+
+  if (!!error || !sponsoredBanners || sponsoredBanners.length === 0) {
+    return null
+  }
+
+  const banner = sponsoredBanners[0]
+  const dataProperties = getDataProperties(banner)
+
+  return (
+    <div
+      {...dataProperties}
+      className={`${handles.bannerWrapper} flex justify-center items-center`}
+      style={{
+        maxWidth: banner.advertisement.width,
+        height: 'auto',
+        aspectRatio: `${styleProps.ratio}`,
+      }}
+    >
+      {loading ? (
+        <Spinner size={styleProps.spinnerSize} />
+      ) : (
+        <button
+          onClick={() => {
+            handleClick(banner.advertisement.targetUrl)
+          }}
```

Comment:
Is there any particular reason to use a <button> tag instead of an <a> tag? I think this might be bad for SEO